File: app/src/main/java/no/joharei/flixr/modules/AndroidModule.java
package no.joharei.flixr.modules;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import no.joharei.flixr.MainApplication;

/**
 * Module for all Android related provisions
 */

@Module
public class AndroidModule {
    @Provides
    @Singleton
    Context provideAppContext() {
        return MainApplication.getInstance().getApplicationContext();
    }

    @Provides
    SharedPreferences provideDefaultSharedPreferences(final Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }
}

File: app/src/main/java/no/joharei/flixr/modules/AndroidModule.kt
package no.joharei.flixr.modules

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import dagger.Module
import dagger.Provides
import no.joharei.flixr.MainApplication
import javax.inject.Singleton

/**
 * Module for all Android related provisions
 */

@Module
class AndroidModule(val app: MainApplication) {
    @Provides
    @Singleton
    fun provideApplication(): MainApplication = app

    @Provides
    @Singleton
    fun provideApplicationContext(): Context = app

    @Provides
    fun provideDefaultSharedPreferences(context: Context): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(context)
    }
}
